src/types/index.ts

```diff
@@ -131,31 +141,100 @@ export enum ProposalType {
   CRITICAL,
 }
 
+export enum VoteSupport {
+  Against = 0,
+  For,
+  Abstain,
+}
+
+type VoterApiResponse = {
+  proposalId: number;
+  address: string;
+  blockNumber: number;
+  blockTimestamp: number;
+  reason: string | undefined;
+  votesMantissa: string;
+  support: VoteSupport;
+};
+
+export type ProposalVoter = Pick<
+  VoterApiResponse,
+  Exclude<keyof VoterApiResponse, 'votesMantissa' | 'blockTimestamp'>
+> & {
+  votesMantissa: BigNumber;
+  blockTimestamp: Date;
+};
+
+export type ForVoter = Pick<ProposalVoter, Exclude<keyof ProposalVoter, 'support'>> & {
+  support: VoteSupport.For;
+};
+
+export type AgainstVoter = Pick<ProposalVoter, Exclude<keyof ProposalVoter, 'support'>> & {
+  support: VoteSupport.Against;
+};
+
+export type AbstainVoter = Pick<ProposalVoter, Exclude<keyof ProposalVoter, 'support'>> & {
+  support: VoteSupport.Abstain;
+};
```

Comment:
If I'm not mistaken, you can use `Omit` to remove some code:

```suggestion
export type ProposalVoter = Omit<VoterApiResponse, 'votesMantissa' | 'blockTimestamp'> & {
  votesMantissa: BigNumber;
  blockTimestamp: Date;
};

export type ForVoter = Omit<VoterApiResponse, 'support'> & {
  support: VoteSupport.For;
};

export type AgainstVoter = Omit<VoterApiResponse, 'support'> & {
  support: VoteSupport.Against;
};

export type AbstainVoter = Omit<VoterApiResponse, 'support'> & {
  support: VoteSupport.Abstain;
};
```